src/Raven.Server/Documents/ShardedHandlers/ShardedSmugglerHandler.cs

```diff
@@ -0,0 +1,375 @@
+                    }
+                    catch (Exception)
+                    {
+                        HttpContext.Abort();
+                    }
+                }
+
+            }
+        }
+
+        public async Task<IOperationResult> ExportShardedDatabaseInternalAsync(
+            DatabaseSmugglerOptionsServerSide options,
+            Action<IOperationProgress> onProgress,
+            BlittableJsonReaderObject blittableJson,
+            JsonOperationContext context,
+            OperationCancelToken token)
+        {
+            var tasks = new List<Task>();
+            var contextList = new List<JsonOperationContext>();
+
+            for (int i = 0; i < ShardedContext.ShardCount; i++)
+            {
+                JsonOperationContext jsonOperationContext;
+                ContextPool.AllocateOperationContext(out jsonOperationContext);
+                contextList.Add(jsonOperationContext);
+            }
+
+            RelativeShardUrl = RelativeShardUrl.Split('?')[0];
+            var tempFiles = new List<string>();
+            for (int i = 0; i < ShardedContext.ShardCount; i++)
+            {
+                var tempFileName = $"{Server.Configuration.Backup.ShardTempPath}\\{Guid.NewGuid()}";
+                tempFiles.Add(tempFileName);
+
+                var tcs = new TaskCompletionSource<object>(TaskCreationOptions.RunContinuationsAsynchronously);
+                var cmd = new ShardedExportCommand(this, async stream =>
+                {
+                    try
+                    {
+                        var fileInfo = new FileInfo(tempFileName);
+                        var directoryInfo = fileInfo.Directory;
+                        if (directoryInfo != null && directoryInfo.Exists == false)
+                            directoryInfo.Create();
+
+                        using (var fileStream = fileInfo.OpenWrite())
+                            await stream.CopyToAsync(fileStream, 8192, new CancellationToken()).ConfigureAwait(false);
+                        tcs.TrySetResult(null);
+
+                    }
+                    catch (Exception e)
+                    {
+                        //if (Logger.IsOperationsEnabled)
+                        //     Logger.Operations("Could not save export file.", e);
+
+                        tcs.TrySetException(e);
+
+                        if (e is UnauthorizedAccessException || e is DirectoryNotFoundException || e is IOException)
+                            throw new InvalidOperationException($"Cannot export to selected path {tempFileName}, please ensure you selected proper filename.", e);
+
+                        throw new InvalidOperationException($"Could not save export file {tempFileName}.", e);
+                    }
+                }, tcs, blittableJson);
+
+                var task = ShardedContext.RequestExecutors[i].ExecuteAsync(cmd, contextList[i]);
+                tasks.Add(task);
+
+            }
+
+            await tasks.WhenAll();
+
```

Comment:
In the meantime, the original request has timed out, because you haven't sent anything